### DISCUSSION TITLE
Suddenly ignoring some frontmatter

### DISCUSSION THREAD

**Discussion by cscherrer** [quarto-dev/quarto-cli#2234] [Q&A] [answered]
Hi, I'm giving a talk **tomorrow**, and I've been preparing it using Quarto. This is my first time using it, and I've been really enjoying it so far. I'm using Quarto from within VS Code, to produce a RevealJS presentation that uses Julia (so it's going through .ipynb).

This morning, the rendered output is suddenly different, and I have no idea why. The frontmatter and CLI call haven't changed. The .scss file still exists, and hasn't changed.

My frontmatter is
```markdown
---
title: "Applied Measure Theory for Composable Statistical Modeling"
author: Chad Scherrer
execute:
    keep-ipynb: true
    keep-md: true
    echo: true
    daemon: 3000
engine: jupyter
format: 
    revealjs:
        logo: logo.png
        slide-number: true
        theme: [default, slides.scss]
        code-fold: false
        code-line-numbers: false
---
```

The (automated) CLI call is
```
quarto preview topos-talk.qmd --to html --no-browser --no-watch-inputs
```

Previously it was (correctly) producing a RevealJS presentation. Now it produces much more generic HTML, just very basic scrollable stuff. Searching that HTML for "reveal" gives no results.

As I said above, my presentation is **tomorrow**. I know this is short notice, but Quarto has been working great for the last week, until just this morning. I'd greatly appreciate if someone could please help me out with this.

Thank you!

**Comment by mcanouil** [comment]
With the command `quarto preview topos-talk.qmd --to html --no-browser --no-watch-inputs` you explicitly ask for the format to be `html`not `revealjs`, thus you get an HTML following the default format.
Try:
```bash
quarto preview topos-talk.qmd --to revealjs --no-browser --no-watch-inputs
```
Or let Quarto do its thing by using your default and only format from the yaml.
```bash
quarto preview topos-talk.qmd --no-browser --no-watch-inputs
```

**Comment by cscherrer** [comment]
That makes sense, I'll try this. But why would the VS Code hotkey suddenly change? (I don't have a way to check the previous CLI calls)

**Reply by mcanouil** [reply]
VSCode default button will use `quarto preview topos-talk.qmd --no-browser --no-watch-inputs`.
But you have access to the main default format, _i.e._, `--to html`, `--to pdf`, and , `--to docx`.
<img width="280" alt="Screenshot of the render button of the Quarto Visual Studio Code extension, in Visual Studio Code showing the different options: 'render', 'render HTML', 'render PDF', and 'render Word'" src="https://user-images.githubusercontent.com/8896044/187705168-2ccc9e24-b225-41a7-b141-ae00d12f6209.png">

PS: if the answer provided above actually solves your issue/question, please select it as the answer to let others know it has been answered.

**Comment by cscherrer** [comment, marked as answer]
From some experimentation, it seems that calling `Render HTML` *redefines the shortcut*. So on subsequent renders, it continues to render HTML. To get back to the YAML-specified settings, you need to call `Render` from VS Code, which will then reset the shortcut back to normal.

Or maybe a clearer way to put it... The shortcut doesn't call `Render`, but instead calls *the render command called most recently*.

This seems like a bug to me, or at least something that could be made more clear in the docs.

In the context of VS Code, I think this is the solution - call Render from the `ctrl-shift-P` options to reset the shortcut. I'll mark it as the solution. Thanks for the quick response!

**Reply by mcanouil** [reply]
Yes, this is exactly it.
If I remember well my time on RStudio, the render button has the same behaviour.

**Reply by mcanouil** [reply]
From the button, you can do that, no need to get to the command palette.

https://user-images.githubusercontent.com/8896044/187707543-2c23ec77-5607-490c-b0bc-4a7c2cb60a81.mov

**Reply by cscherrer** [reply]
One thing that was throwing me is that RevealJS *is* HTML, so I wouldn't expect `--to html` to be a problem.

**Reply by mcanouil** [reply]
`--to` does not define the type of document to be rendered but the format, although "render Word" feels inconsistent with the other options since "Word" is a software, while the format is `docx`.

**Reply by cscherrer** [reply]
Right, I see that now. Just pointing out that this is a subtle point and, at least for me, a source of confusion.

**Reply by mcanouil** [reply]
I made a [PR to the VSCode](https://github.com/quarto-dev/quarto-vscode/pull/70) extension for make the buttons name consistent, i.e., focussing on formats/extension which could alleviate a bit the possible confusion.
